src/core/player.ts

```diff
@@ -87,99 +89,111 @@ export default class Player {
 	/**
 	 * init
 	 * Extends by the provider
+	 * @abstract
 	 */
 	init() {
```

Comment:
On another project I used similar logic in TS and wrote

```suggestion
	protected abstract init() {
```

The `throw` may not be necessary 🤔

Reply:
Yes, `protected abstract init()` is better than `throw` but I get `Abstract methods can only appear within an abstract class. ts(1244)`.
As alternative we can remove the method implementation:
```ts
init(): void { }
```

Reply:
We can add the abstract keyword also on the Player class, the error should disappear